.github/workflows/deploy.yml

```diff
@@ -0,0 +1,65 @@
+name: deploy
+
+"on":
+  push:
+    branches: [main]
+  pull_request:
+    branches: [main]
+  workflow_run:
+    workflows: [ "test" ]
+    types:
+      - completed
+
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check credentials
+        env:
+          STAGING_GCLOUD_PROJECT: ${{ secrets.STAGING_GCLOUD_PROJECT }}
+          STAGING_GCLOUD_SA_KEY: ${{ secrets.STAGING_GCLOUD_SA_KEY }}
+        run: |
+          # Check credentials
+          CREDENTIALS_PRESENT=true
+          function check() {
+            if eval "test -z \"\$$1\""; then
+              echo "$1 secret is missing." >&2
+              CREDENTIALS_PRESENT=false
+            fi
+          }
+          check STAGING_GCLOUD_PROJECT
+          check STAGING_GCLOUD_SA_KEY
+          if "$CREDENTIALS_PRESENT"; then
+            echo "All required credentials are present, continuing." >&2
+          else
+            echo "Required credentials are missing, aborting." >&2
+          fi
+          "$CREDENTIALS_PRESENT"
+      - uses: actions/checkout@v2
+      - name: Set up Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.9
+      - name: Install
+        run: |
+          python -m pip install --upgrade pip setuptools
+          pip3 install --upgrade '.[dev]'
+      - name: Setup Google Cloud SDK
+        uses: google-github-actions/setup-gcloud@master
+        with:
+          project_id: ${{ secrets.STAGING_GCLOUD_PROJECT }}
+          service_account_key: ${{ secrets.STAGING_GCLOUD_SA_KEY }}
+          export_default_credentials: true
```

Comment:
I wish we could reuse all of the above steps as well, instead of copying them, but let's start with the simply copy.